### PR TITLE
✨ Add Core

### LINKED ISSUE #16: Implement core
Core functionality should be:

1. Implement Process (currently it's called Spider, but i think it makes more sense if we call it process since it's a process), for each crawling process.
2. Implement Crawler class
3. Implement the functionality of appending all those process, like a router configuration in fastapi.
4. Initiate all process to rocketry task to write print("Hello fastcrawler") on console, present it to a team member confirming your work in progress.
5. Implement the functionality of making request, in minimal version, no feature for now. let's KISS :)
6. Parse the result and print it on console. Catch all errors in memory for now if there's any.
7. Be friendly to uvicorn (async serve) to run the crawler with uvicorn.

This issue shall be started to work when schedule is merged

### PR DESCRIPTION
**Pull request by ManiMozaffar**
- [x] implement Spider
- [x] implement Crawler
- [x] Connect crawler to rocketry
- [x] implement FastCrawler

closes #16

### CODE REVIEW

File: fastcrawler/engine/aio.py [hunk 1]
```diff
@@ -6,19 +6,23 @@
...
 from aiohttp.cookiejar import Morsel
 
-from fastcrawler.engine.contracts import ProxySetting, Response, SetCookieParam
+from .contracts import ProxySetting, Request, Response, SetCookieParam
 
 
 class AioHttpEngine:
+    default_request_limit = 1
+    request_cls = Request
+    response_cls = Response
```
Comment (aerosadegh) [inline fastcrawler/engine/aio.py:15]: where use the request_cls, response_cls?

File: docs_src/processor/tutorial001.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        assert len(all_data) == 20
+
+
+async def main():
+    process = Process(
+        spider=MySpider(),
+        cond="every 1 second",
+        controller=ProcessController(app=RocketryApplication()),
+    )
+    await process.add_spiders()
```
Comment (aerosadegh) [inline docs_src/processor/tutorial001.py:36]: add_spiders() -> empty?!
where are the spiders to add?

**Review by aerosadegh — APPROVED** [review]
Need Some Changes:
1. add_spiders ? 
2. request_cls using for?

File: docs_src/processor/tutorial001.py (hunk 2 above)
Comment (ManiMozaffar) [reply docs_src/processor/tutorial001.py:36]: we define them in the chain
```python
async def main():
    process = Process(
        spider=MySpider() >> MySpiderTwo() >> MySpiderThree(),
        cond="every 1 second",
        controller=ProcessController(app=RocketryApplication()),
    )
    await process.add_spiders()
 ```
the method `add_spiders ` also adds them to the controller, if there's any defined.
it is not meant to be called like this, usually this happen underhood in FastCrawler App. but if a user want to initiate and run a process manually, that's the way to go.

File: fastcrawler/engine/aio.py (hunk 1 above)
Comment (ManiMozaffar) [reply fastcrawler/engine/aio.py:15]: Nice catch, forgot to use them in code. Will make another commit now.

Comment (ManiMozaffar) [reply fastcrawler/engine/aio.py:15]: self.response_cls

this is already used. when translating response.

**Pull request merged**